### DISCUSSION TITLE
".justify-content-end" to align menu links to right side not working

### DISCUSSION THREAD

**Discussion by sandarbhasthana** [twbs/bootstrap#36773] [Q&A] [answered]
I'm designing a website and just shifted to incorporating Bootstrap 5 and have made a navbar from the Bootstrap Docs and as by default the menu was left aligned right after the brand image.
To change the direction of the menu from left to right, Bootstrap has a .justify-content-end class but adding it to my <ul> has no effect.

Here's my code:

`<nav class="navbar navbar-expand-lg bg-light">
      <div class="container-fluid">
        <div class="nav-img">
          <a class="navbar-brand" href="#">
            <img
              src="https://getbootstrap.com/docs/5.2/assets/brand/bootstrap-logo.svg"
              alt=""
              width="50"
              height="44"
            />
          </a>

          <button
            class="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarSupportedContent"
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span class="navbar-toggler-icon"></span>
          </button>
        </div>
        <div
          class="collapse navbar-collapse"
          id="navbarSupportedContent nav-text"
        >
          <ul class="navbar-nav me-auto mb-2 mb-lg-0 justify-content-end">
            <li class="nav-item">
              <a class="nav-link active" aria-current="page" href="#">Home</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="#">About</a>
            </li>
            <li class="nav-item">
              <a class="nav-link" href="#">Services</a>
            </li>
          </ul>
        </div>
      </div>
    </nav>`

Here's the screenshot

![Screenshot_1](https://user-images.githubusercontent.com/24496556/179757490-483ce7e2-4eec-4713-ac52-5d5ff47fd6b3.png)

**Comment by TravelTimN** [comment, marked as answer]
@sandarbhasthana 
`.justify-content-end` needs to be used in conjunction with `d-flex` for example, as it's part of the Flexbox functionality.

However, perhaps you simply just need to move the `<a><img></a>` down below the other elements (eg: links). Here's a quick example for you:

```html
<nav class="navbar navbar-expand-lg bg-light">
  <div class="container-fluid">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>
    <div class="collapse navbar-collapse" id="navbarNav">
      <ul class="navbar-nav">
        <li class="nav-item">
          <a class="nav-link active" aria-current="page" href="#">Home</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="#">About</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="#">Services</a>
        </li>
      </ul>
    </div>
    <a class="navbar-brand" href="#">
      <img src="https://getbootstrap.com/docs/5.2/assets/brand/bootstrap-logo.svg" alt="" width="30" height="24">
    </a>
  </div>
</nav>
```

Preview:
![image](https://user-images.githubusercontent.com/41592028/179799102-54a2fea1-ed39-49d8-aa00-541de9a94c18.png)


If you also want the links to move to the right-hand side, then yes, this is where you'd add `d-flex justify-content-end` as well.
For example:

```html
<nav class="navbar navbar-expand-lg bg-light">
  <div class="container-fluid">
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>
    <div class="collapse navbar-collapse d-flex justify-content-end" id="navbarNav">
      <ul class="navbar-nav">
        <li class="nav-item">
          <a class="nav-link active" aria-current="page" href="#">Home</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="#">About</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="#">Services</a>
        </li>
      </ul>
    </div>
    <a class="navbar-brand" href="#">
      <img src="https://getbootstrap.com/docs/5.2/assets/brand/bootstrap-logo.svg" alt="" width="30" height="24">
    </a>
  </div>
</nav>
```

Preview:
![image](https://user-images.githubusercontent.com/41592028/179799515-85756044-c568-4da7-9bb8-dbb9b03cdd02.png)

I hope this was helpful?

**Reply by sandarbhasthana** [reply]
@TravelTimN Thanks a lot Tim for the clarification, although I only intend to move the links to the right side and keep the brand on the left, but your detailed answer has shown me how to achieve it🙂
